main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
> Regarding the security concerns - could you give an example please?

@weltling Well, there's more than one syslog daemon (`syslogd`, `slog`, `rsyslog`, `syslog-ng`, etc).  I've not used them all.

Then there's all of the remote servers that the local server could possibly forward messages to, etc. including proprietary loggers in firewalls, authentication servers, etc.

But it's not my allegation that any such vulnerability exists, as I said above.

It was alleged here in the bug report:

```
 [2017-07-05 23:26 UTC] spam2 at rhsoft dot net

> I don't think there is an overall conses on how to handle this, 
> the RFC suggests that a receiver has to deal with that if 
> non-printable (non-allowed) characters are used in the MSG part

yeah, that's how to create security issues as mod_rewrite did in case one did "cat logfile" leading to execute commands because of not properly filtered control chars
```